accord-core/src/main/java/accord/coordinate/FetchData.java

```diff
@@ -415,14 +444,92 @@ public Void apply(SafeCommandStore safeStore)
                     break;
             }
 
+            return updateDurability(safeStore, safeCommand);
+        }
+
+        // if can only propagate Truncated, we might be stale; try to upgrade the
+        private Known applyOrUpgradeTruncated(SafeCommandStore safeStore, SafeCommand safeCommand, Command command)
+        {
+            // if our peers have truncated this command, then either:
+            // 1) we have already applied it locally; 2) the command doesn't apply locally; 3) we are stale; or 4) the command is invalidated
+            if (command.saveStatus().isUninitialised())
+                return null; // TODO (expected): maybe stale?
+
+            if (command.hasBeen(PreApplied))
+            {
+                updateDurability(safeStore, safeCommand);
+                return null;
+            }
+
+            if (Infer.safeToCleanup(safeStore, command, route, full.executeAt))
```

Comment:
I assume you mean in `FetchData`. The premise of first checking if we're uninitialised is to not create a record for something we think needs to perhaps be cleaned up if we don't already have one. But I'm not sure this makes sense anymore - the only case in which this could happen would be if we adopt a new range for e.g. execution after first triggering a `FetchData` for coordination epochs - but in this case we should either have durably erased it or we need to handle it.